### PR TITLE
Refactor: unify Ardana and Crowbar test automation naming scheme (SOC-8475)

### PR DESCRIPTION
**Pull request by stefannica**
The scripts and Jenkins jobs that automate deploying Crowbar in the
ECP, which share a large portion of the Ardana CI code base, still use
`ardana` to name folders, files and runtime artifacts.

This is a refactoring PR which unifies the naming scheme such that
`cloud` is used instead of `ardana` wherever the named artifacts apply
to both Crowbar and Ardana products. The following are covered:

1. a bulk search-and-replace change, replacing `ardana_env` with
`cloud_env`
2. rename the `scripts/jenkins/ardana` folder to `scripts/jenkins/cloud`
3. rename the `openstack-ardana.groovy` library file to `openstack-cloud.groovy` and update its references to reflect that it is a common cloud library
4. unified and improved heat stack naming scheme

    The naming scheme used for the generated heat stacks and
    their resources is changed to reflect the fact that they are used in common
    for both Crowbar and Ardana and to be better aligned with the
    socok8s heat stack naming scheme.
    
    A virtual cloud deployment usually consists of two heat stacks,
    one for the actual cloud setup and the other one for the SES
    cluster used for storage. This is the new naming scheme, based
    on the `cloud_env` parameter value supplied by the user or the
    periodic Jenkins jobs, which uniquely identifies a cloud deployment:

      - the cloud heat stack name will be `<cloud_env>-cloud` and the
        names of its resources will also be prefixed by the same identifier

      - the SES heat stack name will be `<cloud_env>-ses` and the
        names of its resources will also be prefixed by the same identifier

    The heat_stack ansible role will continue to support the old
    naming scheme, to ensure that the existing heat stacks that were created
    using the old naming scheme are properly cleaned up.

### CODE REVIEW

File: scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml [hunk 1]
```diff
@@ -23,6 +23,15 @@
   register: get_stack
   failed_when: false
 
+  # backwards-compatibility: handle old stack naming
```
Comment (nicolasbock) [inline scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml]: Maybe you could add a note that we can remove this after some time, i.e. once we don't have running stacks with the old naming scheme anymore.

File: scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml [hunk 2]
```diff
@@ -32,6 +41,16 @@
   when: heat_action == 'delete'
   failed_when: false
 
+  # backwards-compatibility: handle old stack naming for SES
```
Comment (nicolasbock) [inline scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml]: Same here (regarding the note).

File: scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml (hunk 1 above)
Comment (stefannica) [reply scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml]: I did you one better: I already created a PR for this that we will merge in one week or so: #3490

Comment (nicolasbock) [reply scripts/jenkins/ardana/ansible/roles/heat_stack/tasks/main.yml]: Perfect! :+1:

**Pull request merged**